portfolio/src/main/webapp/script.js

```diff
@@ -12,83 +12,122 @@
-function setToDefault(name, toRemove){
-    var i, elements;
-    elements = document.getElementsByClassName(name);
-    for(i = 0; i < elements.length; i++){
-        elements[i].className = elements[i].className.replace(toRemove, "");
-    }
+/**
+ * Sets className of all elements of a given 
+ * className name from "active" to default "".
+ * @param {name} arg The name of the class I need to modify.
+ * @param {toRemove} arg The string that I need to remove 
+ *  from the class name I'm modifying.
+ */
+replaceClassName = (name, toRemove) => {
+  const elements = document.getElementsByClassName(name);
+  for (let i = 0; i < elements.length; i++) {
+    elements[i].className = elements[i].className.replace(toRemove, "");
+  }
+  // Using forEach made the sections no longer display even when 
+  // I clicked on a tab
+
+  /* elements.forEach((element) => {
+    element.className = element.className.replace(toRemove, "");
+  }) */
 }
 
-/*Changes the current slide in a slideshow by incrementing by the given n*/
-function changeSlide(n){
-    showSlides(slideIndex += n);
+/** 
+ * Changes the current slide in a slideshow 
+ * by incrementing by the given n. 
+ * @param {n} arg The increment/decrement value for showing the next slide.
+ */
+showNextSlide = (n) => {
+  showSlide(slideIndex += n);
 }
 
-/*Display the current nth slide*/
-function currentSlide(n){
-    showSlides(slideIndex = n);
+/** 
+ * Display the current nth slide. 
+ * @param {n} arg The current nth index representing the current slide
+ */
+showCurrentSlide = (n) => {
+  showSlide(slideIndex = n);
 }
 
-/*Event handler that takes care of which slide to display in the slideshow*/
-function showSlides(n){
-    var slides = document.getElementsByClassName("slide");
-    var slideDemos = document.getElementsByClassName("slide-demo");
-    var caption = document.getElementById("caption")
+/** 
+ * Event handler that takes care of which slide to display in the slideshow.
+ * @param {n} arg The variable containing the index for the current nth slide.
+ */
+showSlide = (n) => {
+  const slides = document.getElementsByClassName("slide");
+  const slideDemos = document.getElementsByClassName("slide-demo");
+  const caption = document.getElementById("caption");
 
-    /* Handles n-overflow when n is not in the range of the number of slides or when n is <1*/
-    if(n > slides.length){
-        slideIndex = 1;
-    }
-    if(n < 1){
-        slideIndex = slides.length;
-    }
-    setToNone("slide");
-    setToDefault("slide-demo", " slide-active");
+  // Handles n-overflow when n is not in the range of the number 
+  // of slides or when n is <1.
 
-    slides[slideIndex -1].style.display = "block"
-    slideDemos[slideIndex -1].className += " slide-active"
-    caption.innerHTML = slideDemos[slideIndex -1].alt;
+  if (n > slides.length) {
+    slideIndex = 1;
+  }
+    
+  else if (n < 1) {
+    slideIndex = slides.length;
+  }
+  setToNone("slide");
```

Comment:
Rather than going through all the slides to set their display to none, it would be more efficient to store whichever slide is currently displayed and just make that one less visible.